fullstop-violation/fullstop-violation-jpa/src/main/java/org/zalando/stups/fullstop/violation/service/impl/ApplicationVersionServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package org.zalando.stups.fullstop.violation.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.zalando.stups.fullstop.violation.entity.ApplicationEntity;
+import org.zalando.stups.fullstop.violation.entity.Stack;
+import org.zalando.stups.fullstop.violation.entity.VersionEntity;
+import org.zalando.stups.fullstop.violation.repository.ApplicationRepository;
+import org.zalando.stups.fullstop.violation.repository.VersionRepository;
+import org.zalando.stups.fullstop.violation.service.ApplicationVersionService;
+
+import javax.transaction.Transactional;
+
+import java.util.Optional;
+
+import static javax.transaction.Transactional.TxType.REQUIRES_NEW;
+
+@Service
+public class ApplicationVersionServiceImpl implements ApplicationVersionService {
+
+    @Autowired
+    private ApplicationRepository applicationRepository;
+
+    @Autowired
+    private VersionRepository versionRepository;
+
+    @Override
+    @Transactional(REQUIRES_NEW)
+    public Optional<Stack> saveStack(final String applicationId, final String applicationVersion) {
+
+        if (applicationId == null || applicationVersion == null) {
```

Comment:
In some cases, we only have the app id and not the version (see ScmCommitsJob) and we should try to save at least this single information.